### PR TITLE
update test for shrink removing zero lamports

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
test could add some more variations.

#### Summary of Changes
test several version of latest full snapshot slot.

Fixes #

### CODE REVIEW

File: accounts-db/src/accounts_db.rs [hunk 1]
```diff
@@ -11067,52 +11067,86 @@ pub mod tests {
...
+                .alive_bytes
+                .fetch_sub(aligned_stored_size(0), Ordering::Relaxed);
 
-        // Simulate rooting the zero-lamport account, should be a
-        // candidate for cleaning
-        accounts.calculate_accounts_delta_hash(1);
-        accounts.add_root_and_flush_write_cache(1);
+            if let Some(latest_full_snapshot_slot) = latest_full_snapshot_slot {
+                accounts.set_latest_full_snapshot_slot(latest_full_snapshot_slot);
+            }
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs:11103]: Can we do
```suggestion
            *accounts.latest_full_snapshot_slot.lock_write() = latest_full_snapshot_slot;
```
instead?

Right now the test relies on the `None` case going first. If we always set the latest full snapshot slot directly, the test will be a little more resilient to refactors.

**Review by brooksprumo — COMMENTED** [review]
Code looks good. Just some nits.

File: accounts-db/src/accounts_db.rs [hunk 2]
```diff
@@ -11067,52 +11067,86 @@ pub mod tests {
...
-        // for testing, we need to cause shrink to think this will be productive.
-        // The zero lamport account isn't dead, but it can become dead inside shrink.
-        accounts
-            .storage
-            .get_slot_storage_entry(1)
-            .unwrap()
-            .alive_bytes
-            .fetch_sub(aligned_stored_size(0), Ordering::Relaxed);
+            // Slot 1 should be cleaned, but
+            // zero-lamport account should not be cleaned since latest_full_snapshot_slot is before slot 1
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs]: The latest full snapshot slot can be 2 though, right?

File: accounts-db/src/accounts_db.rs (hunk 1 above)
Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs:11103]: i wanted specifically to test the default behavior case. So if we get refactors, someone has to revisit this.

File: accounts-db/src/accounts_db.rs (hunk 2 above)
Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs]: updated the comments

File: accounts-db/src/accounts_db.rs (hunk 1 above)
Comment (brooksprumo) [reply accounts-db/src/accounts_db.rs:11103]: Ok, that works for me, thanks for the info.

**Review by brooksprumo — APPROVED** [review]
:shipit:

File: accounts-db/src/accounts_db.rs (hunk 1 above)
Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs:11103]: and, actually, it doesn't rely on any ordering (like `None` going first). Test creates a new `AccountsDb` with each value of `latest_full_snapshot_slot`.

**Review by HaoranYi — COMMENTED** [review]
lgtm!

**Pull request merged**